src/main/java/seedu/address/model/transaction/TransactionList.java

```diff
@@ -78,11 +81,14 @@ public void remove(T toRemove) {
         if (!internalList.remove(toRemove)) {
             throw new TransactionNotFoundException();
         }
+        sort();
     }
 
+
     public void setTransactions(TransactionList<T> replacement) {
         requireNonNull(replacement);
         internalList.setAll(replacement.internalList);
+        sort();
```

Comment:
[Suggestion, not sure if it will work]
For all the sort(); here, will it be possible to only call them once, for example under ui/MainWindow.java, under `CommandResult` method. When you finish any command, then call sort() there?

Reply:
This doesn't work because when you do a sort that's not on the original list, the indexes become inconsistent and when the user attempts to remove something it removes the wrong item